tf-psa-crypto/tests/suites/test_suite_psa_crypto.function

```diff
@@ -9818,9 +9838,24 @@ void raw_key_agreement(int alg_arg,
                                      &output_length));
     TEST_MEMORY_COMPARE(output, output_length,
                         expected_output->x, expected_output->len);
+
+    mbedtls_platform_zeroize(output, expected_output->len + 1);
+    output_length = 0;
+
+    psa_set_key_bits(&shared_secret_attributes, (expected_output->len + 1) * 8);
+    PSA_ASSERT(psa_key_agreement(our_key, peer_key_data->x, peer_key_data->len,
+                                 alg, &shared_secret_attributes, &shared_secret_id));
+
+    PSA_ASSERT(psa_export_key(shared_secret_id, output, expected_output->len + 1, &output_length));
```

Comment:
At present this is left to the driver to check that the size in the key attributes match the actual supplied data. This seems to be the way it is [supposed to work in `psa_import_key()`](https://github.com/Mbed-TLS/mbedtls/blob/development/docs/proposed/psa-driver-interface.md#key-size-determination-on-import) so I'm happy to infer that this function is the same. However it would be good to have a more clear documentation of exactly what properties must be checked in drivers vs the core.

Reply:
This is something that `import_key` should handle, don't forget to add negative tests to that too.